pkg/collector/upgrade/upgrade.go

```diff
@@ -133,16 +134,35 @@ func (u VersionUpgrade) ManagedInstance(ctx context.Context, otelcol v1alpha1.Op
 
 	for _, available := range versions {
 		if available.GreaterThan(instanceV) {
-			upgraded, err := available.upgrade(u, &otelcol) //available.upgrade(params., &otelcol)
-
-			if err != nil {
-				u.Log.Error(err, "failed to upgrade managed otelcol instances", "name", otelcol.Name, "namespace", otelcol.Namespace)
-				return otelcol, err
+			if available.upgrade != nil {
```

Comment:
i wonder if we should be recording some metrics about these upgrades:
* how often they succeed / fail
* reason for failure

Reply:
I think this is out of the scope of this PR. We should probably book a separate ticket for this if necessary.